### PR TITLE
Imageplayer keeps player instance on slice position change

### PR DESCRIPTION
**Pull request by woeishi**
was previously discarding preloaded data if dir input changed to another slice

### CODE REVIEW

File: vvvv45/src/nodes/plugins/Texture/ImagePlayer/PlayerNode.cs [hunk 1]
```diff
@@ -114,8 +114,22 @@ public void Evaluate(int spreadMax)
                 .CombineWith(FPreloadFramesIn)
                 .CombineWith(FReloadIn);
             
-            FImagePlayers.Resize(spreadMax, CreateImagePlayer, DestroyImagePlayer);
-            
+            var oldPlayers = new List<ImagePlayer>(FImagePlayers.ToList());
```
Comment (azeno) [inline vvvv45/src/nodes/plugins/Texture/ImagePlayer/PlayerNode.cs]: var oldPlayers = FImagePlayers.ToList();

File: vvvv45/src/nodes/plugins/Texture/ImagePlayer/PlayerNode.cs [hunk 2]
```diff
@@ -114,8 +114,22 @@ public void Evaluate(int spreadMax)
...
                 .CombineWith(FReloadIn);
             
-            FImagePlayers.Resize(spreadMax, CreateImagePlayer, DestroyImagePlayer);
-            
+            var oldPlayers = new List<ImagePlayer>(FImagePlayers.ToList());
+
+            FImagePlayers.SliceCount = spreadMax;
+            for (int i = 0; i < spreadMax; i++)
+            {
+                var player = oldPlayers.Where(p => p.Directories.SpreadEqual(FDirectoryIn[i])).DefaultIfEmpty(null).First();
```
Comment (azeno) [inline vvvv45/src/nodes/plugins/Texture/ImagePlayer/PlayerNode.cs]: var player = oldPlayers.FirstOrDefault(p => p.Directories.SpreadEqual(FDirectoryIn[i]));

**Review by azeno — CHANGES_REQUESTED** [review]
Just a few minor code simplifications. Otherwise it looks good.

**Comment by woeishi** [comment]
here you go

 List<ImagePlayer> oldPlayers =... because vs was complaining in the foreach below that ? doesn't implement GetEnumerator

**Comment by azeno** [comment]
Thanks!

**Pull request merged**